herddb-collections/src/main/java/herddb/collections/CollectionsManager.java

```diff
@@ -0,0 +1,331 @@
+        return new Builder();
+    }
+
+    private CollectionsManager(long maxMemory, Path tmpDirectory) {
+        ServerConfiguration configuration = new ServerConfiguration();
+        
+        configuration.set(ServerConfiguration.PROPERTY_MEMORY_LIMIT_REFERENCE, maxMemory);
+        
+        server = new DBManager("localhost",
+                new MemoryMetadataStorageManager(),
+                new FileDataStorageManager(tmpDirectory,
+                        tmpDirectory, 0, false /* fsync */,
+                        false /* o_direct */, false /* o_direct */, NullStatsLogger.INSTANCE),
+                new MemoryCommitLogManager(), tmpDirectory,
+                new ServerHostData("localhost", 0, "", false, Collections.emptyMap()),
+                configuration, NullStatsLogger.INSTANCE);
+        server.setMaxDataUsedMemory(maxMemory);
+        valuesSerializer = new Serializer() {
+            @Override
+            public VisibleByteArrayOutputStream serialize(Object object) throws Exception {
+                VisibleByteArrayOutputStream oo = new VisibleByteArrayOutputStream();
+                try (ObjectOutputStream ooo = new ObjectOutputStream(oo)) {
+                    ooo.writeUnshared(object);
+                }
+                return oo;
+            }
+
+            @Override
+            public Object deserialize(Bytes bytes) throws Exception {
+                SimpleByteArrayInputStream oo = new SimpleByteArrayInputStream(bytes.getBuffer(), bytes.getOffset(),
+                        bytes.getLength());
+                try (ObjectInputStream ooo = new ObjectInputStream(oo)) {
+                    return ooo.readUnshared();
+                }
+            }
+
+        };
+    }
+
+    public void start() throws Exception {
+        server.start();
+        server.waitForTablespace(TableSpace.DEFAULT, 60000);
+    }
+
+    @Override
+    public void close() {
+        server.close();
+    }
+
+    public static final class Builder {
+
+        private long maxMemory = ServerConfiguration.PROPERTY_MEMORY_LIMIT_REFERENCE_DEFAULT;
+        private Path tmpDirectory;
+
+        public Builder maxMemory(long maxMemory) {
+            this.maxMemory = maxMemory;
+            return this;
+        }
+
+        public Builder tmpDirectory(Path tmpDirectory) {
+            this.tmpDirectory = tmpDirectory;
+            return this;
+        }
+
+        public CollectionsManager build() {
+            return new CollectionsManager(maxMemory, tmpDirectory);
+        }
+
+    }
+
+    public <V> TmpMap<Integer, V> createIntKeyedMap() {
+        final TableSpaceManager tableSpaceManager = server.getTableSpaceManager(TableSpace.DEFAULT);
+        final String tmpTableName = "tmp" + tableNameGenerator.incrementAndGet();
+        Table table = Table
+                .builder()
+                .name(tmpTableName)
+                .column("pk", ColumnTypes.INTEGER)
+                .column("v", ColumnTypes.BYTEARRAY)
```

Comment:
Todo: this table schema is not really needed
As we are using the low level APIs which deal only with pairs of byte[]